#! /bin/sh -e

# All lines beginning with `# DP:' are a description of the patch.
# DP: avoid linking libmachuser and libhurduser against libc

if [ $# -ne 2 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch) patch -d "$2" -f --no-backup-if-mismatch -p0 < $0;;
    -unpatch) patch -d "$2" -f --no-backup-if-mismatch -R -p0 < $0;;
    *)
	echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
	exit 1
esac
exit 0

--- sysdeps/mach/hurd/Makefile	Tue Dec 22 19:15:49 1998
+++ sysdeps/mach/hurd/Makefile	Sun Dec 20 12:08:05 1998
@@ -116,6 +116,8 @@
 	       $(common-objpfx)hurd/libhurduser.so
 $(common-objpfx)libc.so: $(rpcuserlibs)
 rpath-dirs += mach hurd
+LDFLAGS-machuser.so = -nostdlib -nostartfiles
+LDFLAGS-hurduser.so = -nostdlib -nostartfiles
 
 # And get them into the libc.so ldscript.
 $(inst_libdir)/libc.so: $(rpcuserlibs)
